packages/cli/src/utils/input.ts

```diff
@@ -49,6 +25,199 @@ export function parseArgs(args: string): string[] | never {
   }
 }
 
+function quoteArguments(args: string) {
+  const START_LOOKBEHIND = '(?<=^|([,[]\\s*))'; // Character before match is start of line, comma or opening bracket.
+  const END_LOOKAHEAD = '(?=$|([,\\]]\\s*))'; // Character after match is end of line, comma or closing bracket.
+
+  // Replace non quoted hex string by quoted hex string.
+  const MATCH_HEX = new RegExp(
+    START_LOOKBEHIND + '(0[xX][0-9a-fA-F]+)' + END_LOOKAHEAD,
+    'g',
+  );
+  args = args.replace(MATCH_HEX, '"$2"');
+
+  // Replace scientific notation numbers by regular numbers.
+  const MATCH_SCIENTIFIC = new RegExp(
+    START_LOOKBEHIND +
+      '(\\s*[-]?\\d+(\\.\\d+)?e(\\+)?\\d+\\s*)' +
+      END_LOOKAHEAD,
+    'g',
+  );
+  args = args.replace(MATCH_SCIENTIFIC, val => `${new BN(val).toString(10)}`);
+
+  // Replace non quoted number by quoted number.
+  const MATCH_WORDS = new RegExp(
+    START_LOOKBEHIND + '([-]?\\w+)' + END_LOOKAHEAD,
+    'g',
+  );
+  args = args.replace(MATCH_WORDS, '"$2"');
+  return args;
+}
+
+export function parseArg(input: string | string[], type: string): any {
+  const TRUE_VALUES = ['y', 'yes', 't', 'true', '1'];
+  const FALSE_VALUES = ['n', 'no', 'f', 'false', '0'];
+  const ARRAY_TYPE_REGEX = /(.+)\[\d*\]$/; // matches array type identifiers like uint[] or byte[4]
+
+  // TODO: Handle tuples in ABI specification
+
+  // Integers: passed via bignumber to handle signs and scientific notation
+  if (
+    (type.startsWith('uint') ||
+      type.startsWith('int') ||
+      type.startsWith('fixed') ||
+      type.startsWith('ufixed')) &&
+    requireInputString(input)
+  ) {
+    const parsed = new BN(input);
+    if (parsed.isNaN())
+      throw new Error(`Could not parse '${input}' as ${type}`);
+    return parsed.toString(10);
+  }
+
+  // Booleans: match against true, t, yes, 1, etc
+  else if (type === 'bool' && requireInputString(input)) {
+    const lowercaseInput = input.toLowerCase();
+    if (TRUE_VALUES.includes(lowercaseInput)) return true;
+    else if (FALSE_VALUES.includes(lowercaseInput)) return false;
+    else throw new Error(`Could not parse boolean value ${input}`);
+  }
+
+  // Arrays: recursively parse as needed
+  else if (type.match(ARRAY_TYPE_REGEX)) {
```

Comment:
This one should be the first case to try to match. Otherwise, `startsWith(uint)` will match `uint[]`.